src/app/loadout-builder/process.ts

```diff
@@ -416,19 +416,40 @@ export function generateMixesFromPerks(
   return mixes;
 }
 
-function getBaseStatValue(stat: DimStat, item: DimItem) {
-  let baseStatValue = stat.value;
+function getPlugHashesFromCategory(category: DimSocketCategory) {
+  return category.sockets
+    .map((socket) => {
+      return idx(socket, (socket) => socket.plug.plugItem.hash) || null;
+    })
+    .filter(Boolean);
+}
+
+function getBaseStatValues(stats: Dictionary<DimStat>, item: DimItem) {
+  const baseStats = {};
+  _.forEach(statValues, (statHash) => (baseStats[statHash] = stats[statHash].value));
 
   // Checking energy tells us if it is Armour 2.0
   if (item.isDestiny2() && item.sockets && item.energy) {
+    const masterworkSocketCategory = item.sockets.categories.find((category) => {
+      return category.category.categoryStyle === DestinySocketCategoryStyle.EnergyMeter;
+    });
+    const masterworkSocketHashes =
+      (masterworkSocketCategory && getPlugHashesFromCategory(masterworkSocketCategory)) || [];
+
     for (const socket of item.sockets.sockets) {
-      if (socket.plug && socket.plug.stats && socket.plug.stats[stat.statHash]) {
-        baseStatValue -= socket.plug.stats[stat.statHash];
+      const plugHash = idx(socket, (socket) => socket.plug.plugItem.hash) || null;
+
+      if (socket.plug && socket.plug.stats && !masterworkSocketHashes.includes(plugHash)) {
+        for (const statHash of statValues) {
+          if (socket.plug.stats[statHash]) {
+            baseStats[statHash] -= socket.plug.stats[statHash];
+          }
+        }
       }
     }
   }
-
-  return baseStatValue;
+  // mapping out from stat values to ensure ordering
+  return _.map(statValues, (statHash) => baseStats[statHash]);
```

Comment:
If `statValues` is an array, you can just use `statValues.map(...)`

Reply:
done